systemvm/patches/debian/config/opt/cloud/bin/cs/CsAddress.py

```diff
@@ -659,7 +663,10 @@ def is_guest_gateway(self, bag, ip):
         gw = interface.get_gateway()
         logging.info("Interface has the following gateway ==> %s", gw)
 
-        if bag['nw_type'] == "guest" and rip == gw:
+        guest_gw = self.config.cmdline().get_guest_gw()
+        logging.info("Interface has the following gateway ==> %s", guest_gw)
+
+        if bag['nw_type'] == "guest" and (rip == gw or rip == guest_gw):
```

Comment:
I find that makes the code a bit harder to read if we did that. Then it is not as obvious you are returning a boolean. This stuff is already complicated, for two lines of code, I don't think it makes sense to make it harder to understand. :)